config/application.yml.example

```
@@ -27,6 +27,7 @@ DISABLE_COI_AUDIT_DATE_VALIDATION: 'false'
 DISABLE_FINDING_FINAL_STATE_ROLE_VALIDATION: 'false'
 DISABLE_FINDINGS_EXPIRATION_NOTIFICATION: 'false'
 DISABLE_REVIEW_AUDITED_VALIDATION: 'false'
+ENABLE_LIMIT_COMMITMENT_DATE: '{"limit_date": "true", "message_one_year": "Una observación de riesgo alto, no puede tener una fecha estimada de solución mayor a 1 año."}'
```

Esta variable la armaría como sigue:
```suggestion
COMMITMENT_DATE_LIMITS: '{}'
```

Y la estructura, después pasala a JSON, pero sería:

```ruby
{
  first_date: {
    high: {
      '1.year' => 'Una observación de riesgo alto, no puede tener una fecha estimada de solución mayor a 1 año.'
    }
  },
  reschedule: {
    default: {
      '6.months' => 'no se puede reprogramar con una fecha mayor a 6 meses'
    }
  }
}
```

Luego en dentro de la lógica buscás "si el follow_up_date es blank? y hay un key definido en el first_date, que después tiene un key que coincide con el nombre del riesgo y/o default" tenés en el key de esa respuesta lo que tenés que pasar al eval y hacés algo como:

```ruby
if commitment_date > eval(value).from_now.to_date
```

Y algo similar pero comparando con el follow_up_date, cuando el estado es "En proceso de implementación" y las entradas en esa otra key.